### PR TITLE
Fix Array to string conversion in Admin Contacts Modal

### PR DESCRIPTION
**Pull request by julienV**
just a little modification to make it consistent with filter.category_id in getListQuery method

---
#### Steps to reproduce the issue

use a module that uses the backend model with several categories
#### Expected result

there shouldn't be any notice
#### Actual result

there is a notice
Notice: Array to string conversion in <root>/administrator/components/com_contact/models/contacts.php on line 134
#### System information (as much as possible)

joomla staging branch
#### Additional comments

it's just a samll inconsistency in backend models contact.
The filter category_id is expected to be a numeric or an array in getListQuery function, but the store id function assumes it's a string.
hence, setting the filter to an array triggers a warning:

Notice: Array to string conversion in <root>/administrator/components/com_contact/models/contacts.php on line 134

### CODE REVIEW

**Comment by zero-24** [comment]
@julienV can you give us some example code? If this is a real issue we need to fix it also for other models e.g. article ;)

**Comment by julienV** [comment]
you are right, it's the same in articles model indeed...

it happens if you try to reuse the model, in a 3rd party module for example:
$model = JModelLegacy::getInstance('Contacts', 'ContactModel', array('ignore_request' => true));
$model->setState('filter.category_id', array(1, 2 ,3));
$result = $model->getItems();

**Comment by julienV** [comment]
the goal of the getStoreId function is to cache same requests, so another solution could be to generalize the string contruction like this too: 
$id .= serialize($this->getState('filter.category_id'));

**Comment by euismod2336** [comment]
I have tested this item :white_check_mark: successfully on 3e49ad7b5fed0774fb35dcb394d64f1c5a783634<br /><br />Notice before patch, no notice after patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7619">issues.joomla.org/tracker/joomla-cms/7619</a>.</sub>

**Comment by marrouchi** [comment]
@julienV  I would recommend using serialize() like in BannersModelBanners (@package Joomla.Site).

Also, it appears that we have the same issue in ContentModelArticles and BannersModelBanners (@package Joomla.Administrator) models.

**Comment by franz-wohlkoenig** [comment]
@julienV can you please give Test Instructions?

**Comment by jeckodevelopment** [comment]
@julienV please look at the conflicting file

**Comment by julienV** [comment]
@jeckodevelopment i updated the push, and followed the recommandations of @marrouchi to use serialize as it's shorter and covers all situation. And indeed, there was the same issue with content model as well.

File: administrator/components/com_content/models/articles.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ protected function getStoreId($id = '')
 		$id .= ':' . $this->getState('filter.search');
 		$id .= ':' . $this->getState('filter.access');
 		$id .= ':' . $this->getState('filter.published');
-		$id .= ':' . $this->getState('filter.category_id');
+        $id .= ':' . serialize($this->getState('filter.category_id'));
```
Comment (Quy) [inline administrator/components/com_content/models/articles.php]: Fix indentation. Use tabs.

**Comment by julienV** [comment]
i fixed the indentation

**Comment by Schmidie64** [comment]
@icampus 
could you please add a Test Instructions?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7619">issues.joomla.org/tracker/joomla-cms/7619</a>.</sub>

**Comment by SamuelSchepp** [comment]
I have tested this item :white_check_mark: successfully on 5c9f983f9f2143c16bbaafca08fcc5dc32f91714<br /><br />Tested:
Filtered in com_contact Cantacts view for multiple categories.

Without patch:
Said notices appear.

With patch:
No notices appear.

@icampus
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7619">issues.joomla.org/tracker/joomla-cms/7619</a>.</sub>

**Comment by Heggi93** [comment]
@icampus
could you please add more detailed Test Instructions?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7619">issues.joomla.org/tracker/joomla-cms/7619</a>.</sub>

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on 5c9f983f9f2143c16bbaafca08fcc5dc32f91714<br /><br />Solves the Array to string conversion notice after applying patch. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/7619">issues.joomla.org/tracker/joomla-cms/7619</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests. @julienV please resolve conflicting Files.

**Comment by zero-24** [comment]
@franz-wohlkoenig please keep off the RTC label until we have conflicting files and travis or drone errors.

**Comment by ghost** [comment]
will do and hopefully not forget to set RTC later.

**Comment by zero-24** [comment]
👍

**Comment by ghost** [comment]
@julienV can you please resolve conflicting Files?

**Comment by julienV** [comment]
@franz-wohlkoenig sorry for delay, i merged staging.

**Comment by ghost** [comment]
Thanks @julienV

**Pull request merged**